Src/VimCore/Modes_Insert_InsertMode.fs

```diff
@@ -816,6 +831,12 @@ type internal InsertMode
         _sessionData <- { _sessionData with ActiveEditItem = if _isReplace then ActiveEditItem.OverwriteReplace else ActiveEditItem.Paste }
         ProcessResult.Handled ModeSwitch.NoSwitch
 
+    /// Start a undo session in insert mode
+    member x.ProcessUndoStart keyInput =
+        x.CancelWordCompletionSession()
+        _sessionData <- { _sessionData with ActiveEditItem = ActiveEditItem.Undo }

```

Comment:
How does the transaction get reset in this case? Believe that is necessary in order to implement the command properly. Is it done incidentally as a part of handling this.

Reply:
This is in support of `<C-g>u` command (`:help i_Ctrl-G_u` in vim). There are (at least) two commands related to undo in vim, only the first of which I implemented.

In this context `ProcessUndoStart` means "process the first key of a multi-key sequence related to undo. Perhaps the name `ProcessUndoStart`  sounds like it is undoing something but in reality it is just the intermediate state halfway through responding to `<C-g>` followed by some key.

As far as the undo transaction, it will get closed out and reopened once the `u` key is typed, all without ever leaving insert mode.